run.sh

```diff
@@ -66,54 +72,71 @@ loadFullShape() {
   "$SCRIPT_PATH"/load-graph.js "$1" | "$SCRIPT_PATH"/pretty-print.js --prefixes "${prefixes[@]}"
 }
 
+# Semaphore function to limit parallel jobs
+semaphore() {
+  while [ "$CURRENT_JOBS" -ge "$MAX_JOBS" ]; do
+    wait -n
+    CURRENT_JOBS=$((CURRENT_JOBS - 1))
+  done
+  CURRENT_JOBS=$((CURRENT_JOBS + 1))
+}
+
 # iterate over valid cases, run validation and monitor exit code
 for file in $validCases; do
-  name=$(basename "$file")
-  relativePath=$(node -e "console.log(require('path').relative('$WORKING_DIR', '$file'))")
-
-  # check if filter is set and skip if not matching
-  if [ -n "$filter" ] && ! echo "$file" | grep -iq "$filter"; then
-    echo "ℹ️SKIP - $relativePath"
-    continue
-  fi
-
-  {
-    sh -c "$command" > "$file.log" 2>&1
-    success=$?
-  } < "$file"
-
-  if [ $success -ne 0 ] ; then
-    "$SCRIPT_PATH"/report-failure.sh "$file" "$(loadFullShape "$shapesPath")" "$(cat "$file")"
-    FAILED=1
-  else
-    echo "✅ PASS - $relativePath"
-  fi
+  semaphore
+  (
+    name=$(basename "$file")
+    relativePath=$(node -e "console.log(require('path').relative('$WORKING_DIR', '$file'))")
+
+    # check if filter is set and skip if not matching
+    if [ -n "$filter" ] && ! echo "$file" | grep -iq "$filter"; then
+      echo "ℹ️SKIP - $relativePath"
+      exit 0
+    fi
+
+    {
+      sh -c "$command" > "$file.log" 2>&1
+      success=$?
+    } < "$file"
+
+    if [ $success -ne 0 ] ; then
```

Comment:
I tried to run it using the following command:

```sh
npx . --shapes="../cube-link/validation/$1.ttl" \
 --valid-cases="../cube-link/test/$1/valid*.ttl" \
 --invalid-cases="../cube-link/test/$1/invalid*.ttl" \
 --command="npx b59 cube check-metadata --profile" \
 --prefixes=schema,cube=https://cube.link/ \
 --concurrency=8 --debug
```

(I cloned the `cube-link` repo next to the `shacl-test` repo).

And here are the errors I can see from my side (translated from French):

```
line 111: ../cube-link/test//valid*.ttl: No such file or directory
line 105 : [:  : integer expression expected
```

So am I doing something wrong while invocating the command?
Could you provide a working example, so that I can try it?

I took some inspiration from here: https://github.com/zazuko/cube-link/blob/7c2e6db06051b2234d85b82116dd6d16f00c62d5/test/check-metadata.sh

Also, is there any reason to keep this as a shell script instead of a NodeJS script?
That way this would be able to work on more systems, including ones that doesn't bring bash by default.